### PR TITLE
Superb guide: use hooks to make editors way more readable

### PR DESCRIPTION
**Pull request by GregWeil**
`useCurrentUser()` `useNotifications()` `useErrorHandlers()` and `useUploader()` are all equivalent to the their `<BlahConsumer>` versions but don't require tons of nesting to use (you can't use hooks in class components though, so there is still one layer of wrapping to gather them up)

I also cleaned up some props that are marked as optional but aren't.

### CODE REVIEW

File: src/presenters/error-handlers.jsx [hunk 1]
```diff
@@ -1,7 +1,4 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-
-import { NotificationConsumer } from './notifications';
+import { useNotifications } from './notifications';
```
Comment (clottman) [inline src/presenters/error-handlers.jsx]: @ttseng if this PR gets in before yours, you'll probably have merge conflicts again in this file. Let me know if you want help sorting those out.

**Comment by clottman** [comment]
@GregWeil I think you're a few commits behind on Github - I see this line in current-user.jsx on Glitch but not here: 
```
export const useCurrentUser = () => React.useContext(Context);
```

File: src/presenters/project-editor.jsx [hunk 2]
```diff
@@ -55,31 +55,24 @@ ProjectEditor.defaultProps = {
...
-            {children}
-          </ProjectEditor>
-        )}
-      </CurrentUserConsumer>
-    )}
-  </ErrorHandlers>
-);
+const ProjectEditorContainer = ({ api, children, initialProject }) => {
+  const { currentUser } = useCurrentUser();
+  const wrapErrors = useErrorHandlers();
```
Comment (clottman) [inline src/presenters/project-editor.jsx]: While we're here should we rename this variable to be `errorFuncs` or something more descriptive & consistent with the other places we use `useErrorHandlers`?

**Comment by clottman** [comment]
Approved once this PR gets up to date with what's on Glitch. :D This is beautiful.

**Comment by GregWeil** [comment]
> 
> 
> @GregWeil I think you're a few commits behind on Github - I see this line in current-user.jsx on Glitch but not here:
> 
> ```
> export const useCurrentUser = () => React.useContext(Context);
> ```

I actually snuck this in a pr a few days ago, but only used it in the one place
https://github.com/FogCreek/Glitch-Community/pull/261

**Pull request merged**